### DISCUSSION TITLE
CustomDifficulty plugin does not run.

### DISCUSSION THREAD

**Discussion by wwa171** [Pryaxis/TShock#2286] [Q&A]
https://github.com/tanpro260196/CustomDifficulty
I downloaded and tried running it, but an error pops up.

Is there a solution?

`[04/25/21 15:27:53] [Server API] Error: Startup aborted due to an exception in the Server API initialization:
System.InvalidOperationException: Plugin "CustomDifficulty" has thrown an exception during initialization. ---> System.MissingMethodException: Method not found: 'TShockAPI.ConfigFile TShockAPI.TShock.get_Config()'.
   at Koishi.CDMain.Initialize()
   at TerrariaApi.Server.ServerApi.LoadPlugins() in D:\a\TShock\TShock\TerrariaServerAPI\TerrariaServerAPI\TerrariaApi.Server\ServerApi.cs:line 349
   --- End of inner exception stack trace ---
   at TerrariaApi.Server.ServerApi.LoadPlugins() in D:\a\TShock\TShock\TerrariaServerAPI\TerrariaServerAPI\TerrariaApi.Server\ServerApi.cs:line 354
   at TerrariaApi.Server.ServerApi.Initialize(String[] commandLineArgs, Main game) in D:\a\TShock\TShock\TerrariaServerAPI\TerrariaServerAPI\TerrariaApi.Server\ServerApi.cs:line 117
   at TerrariaApi.Server.HookManager.InitialiseAPI() in D:\a\TShock\TShock\TerrariaServerAPI\TerrariaServerAPI\TerrariaApi.Server\HookManager.cs:line 27`

**Comment by hakusaro** [comment]
Unfortunately, this error occurs when you've downloaded an outdated plugin, typically. It means that the author has to update their plugin for the latest version of TShock. The `Method not found` part indicates that they're trying to access something that doesn't exist anymore in TShock. This is because a change was made, and a replacement has been provided, but unfortunately, that plugin doesn't use the replacement.

Your best bet is to humbly ask @tanpro260196 to change their call to the newer version or to ask someone to update it for you. I'm really sorry.

**Reply by wwa171** [reply]
> Unfortunately, this error occurs when you've downloaded an outdated plugin, typically. It means that the author has to update their plugin for the latest version of TShock. The `Method not found` part indicates that they're trying to access something that doesn't exist anymore in TShock. This is because a change was made, and a replacement has been provided, but unfortunately, that plugin doesn't use the replacement.
> 
> Your best bet is to humbly ask @tanpro260196 to change their call to the newer version or to ask someone to update it for you. I'm really sorry.

No need to be sorry, thank you for your reply

**Reply by hakusaro** [reply]
Nah, @wwa171 we've had an hour long discussion in Discord about how terrible of an experience getting this error is with no context. I'm sorry that we've left you out in the cold here. Thank you for asking us and I'm sorry that you're stuck :( ❤️
